cmd/init-osm-controller/init-osm-controller.go

```diff
@@ -0,0 +1,110 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"os"
+	"time"
+
+	"github.com/spf13/pflag"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openservicemesh/osm/pkg/apis/config/v1alpha1"
+	"github.com/openservicemesh/osm/pkg/cli"
+	configClientset "github.com/openservicemesh/osm/pkg/gen/client/config/clientset/versioned"
+	"github.com/openservicemesh/osm/pkg/logger"
+	"github.com/openservicemesh/osm/pkg/version"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const meshConfigName = "osm-mesh-config"
+
+var settings = cli.New()
+
+var (
+	flags = pflag.NewFlagSet("init-osm-controller", pflag.ExitOnError)
+	log   = logger.New("osm-controller/main")
+
+	meshName           string // An ID that uniquely identifies an OSM instance
+	kubeConfigFilePath string
+	osmNamespace       string
+
+	fetchMeshConfigMaxRetry      = 5
+	fetchMeshConfigRetryInterval = 5
+)
+
+func init() {
+	if err := logger.SetLogLevel("info"); err != nil {
+		log.Fatal().Err(err).Msg("Error setting log level")
+	}
+
+	flags.StringVar(&meshName, "mesh-name", "", "OSM mesh name")
+	flags.StringVar(&kubeConfigFilePath, "kubeconfig", "", "Path to Kubernetes config file.")
+	flags.StringVar(&osmNamespace, "osm-namespace", "", "Namespace to which OSM belongs to.")
+}
+
+func parseFlags() error {
+	if err := flags.Parse(os.Args); err != nil {
+		return err
+	}
+	_ = flag.CommandLine.Parse([]string{})
+	return nil
+}
+
+func main() {
```

Comment:
I would try to pull as much of this out into its own function as possible so it can be unit tested.